use-cases.md

```diff
@@ -0,0 +1,144 @@
+    span.set_tag('http.url', request.full_url)
+```
+
+The `operation` above refers to the name the server wants to give to the Span. For example, if the HTTP request was a POST against `/save_user/123`, the operation name can be set to `post:/save_user/`. OpenTracing API does not dictate how application calls the spans.
+
+The `set_tag` calls are examples of recording additional information in the span about the request.
+
+#### In-Process Request Context Propagation
+
+Request context propagation refers to application's ability to associate a certain _context_ with the incoming request such that this context is accessible in all other layers of the application within the same process. It can be used to provide application layers with access to request-specific values such as the identity of the end user, authorization tokens, and the request's deadline. It can also be used for transporting the current tracing Span.
+
+The methods of request context propagation are outside the scope of the OpenTracing API, but it is worth mentioning them here to better understand the following sections. There are two commonly used methods of propagation:
+
+  1. Explicit propagation - the application code is structured to pass around a certain context object. For example, see https://blog.golang.org/context.
+  1. Implicit propagation - the context is stored in platform-specific storage that allows it to be retrieved from any place in the application. Often used by RPC frameworks by utilizing such mechanisms as thread-local or continuation-local storage, or even global variables (in case of single-threaded processes).
+
+### Tracing Client Calls
+
+When an application acts as an RPC client, it is expected to start a new tracing Span before making an outgoing request, and serialize the new span's Trace Context into that request. The following example shows how it can be done for an HTTP request. 
+
+```python
+    def traced_request(request, operation, http_client):
+        # retrieve current span from propagated request context
+        parent_span = get_current_span()
+
+        # start a new child span or a brand new trace if no parent
+        if parent_span is None:
+            span = tracer.start_trace(operation_name=operation)
+        else:
+            span = parent_span.start_child(operation_name=operation)
+        span.set_tag('server.http.url', request.full_url)
+
+        # encode Trace Context into HTTP request headers
+        h_ctx, h_attr = tracer.trace_context_to_text(
+            trace_context=span.trace_context)
+
+        for key, value in h_ctx.iteritems():
+            request.add_header(key, value)
+        if h_attr:
+            for key, value in h_attr.iteritems():
+                request.add_header(key, value)
+
+        # define a callback where we can finish the span 
+        def on_done(future):
+            if future.exception():
+                span.log_event_with_payload('exception', exception)
```

Comment:
I know there is still some debate on the `Error` event spec in https://github.com/opentracing/opentracing-go/pull/38 , but if we do end up with that included, this is a great place to show it off.